custom/icds/views/custom_sms_report.py

```diff
@@ -22,22 +26,49 @@ def dispatch(self, *args, **kwargs):
 
     @property
     def page_context(self):
-        self.request_form = CustomSMSReportRequestForm()
-        return {
-            'request_form': self.request_form,
-        }
+        report_ids = self._get_active_report_ids()
+        report_count = len(report_ids)
+        context = {}
+        disable_submit = report_count >= MAX_CONCURRENT_SMS_REPORTS_ALLOWED
+        if disable_submit:
+            messages.error(self.request, _('Only {max_report_count} concurrent reports are allowed at a time.\
+                Please wait for them to finish.').format(
+                max_report_count=MAX_CONCURRENT_SMS_REPORTS_ALLOWED
+            ))
+        if report_count > 0:
+            context.update({
+                'reports_in_progress': ',  '.join(report_ids)
+            })
+        self.request_form = CustomSMSReportRequestForm(disable_submit=disable_submit)
+        context.update({
+            'request_form': self.request_form
+        })
+        return context
+
+    @memoized
+    def _get_active_report_ids(self):
+        report_tracker = CustomSMSReportTracker(self.request.domain)
+        return report_tracker.active_reports
 
     def post(self, request, *args, **kwargs):
+        report_tracker = CustomSMSReportTracker(request.domain)
+        reports_in_progress = report_tracker.active_reports
+        report_count = len(reports_in_progress)
+
         self.request_form = CustomSMSReportRequestForm(request.POST)
-        user_email = request.user.email
-        if not user_email:
-            messages.error(self.request, _("Unable to find any email associated with your account"))
-            return self.get(*args, **kwargs)
-        if self.request_form.is_valid():
+
+        if self.request_form.is_valid() and report_count < MAX_CONCURRENT_SMS_REPORTS_ALLOWED:
```

Comment:
nit: the ideal thing would be to have a method that does that length comparison and gives back a boolean. You already have this comparison at two places, so its less number of places, as of now.

Reply:
Done.